qiskit_experiments/framework/base_experiment.py

```diff
@@ -238,6 +261,11 @@ def run_analysis(
 
         See :meth:`BaseAnalysis.run` for additional information.
 
+        .. deprecated:: 0.2.0
```

Comment:
We should also deprecate `analysis` option from `BaseExperiment.run` because
- kind of edge case; if analysis is `None` and `run(analysis=True)`.
- now `BaseExperiment.analysis` returns an instance so arg name is really confusing, we may want to set analysis instance rather than bool.

Reply:
I could change it to `analysis="default"` as the kwarg, and allow you to pass in a full on analysis class if you wanted as the runtime option (since this could override the stored analysis class for that execution). And of course then you could still do `analysis=None/False` to not run analysis.